packages/rule-no-broken-links/src/rule.ts

```diff
@@ -0,0 +1,126 @@
+import {
+    IRule,
+    ElementFound,
+    RuleMetadata,
+    IAsyncHTMLElement
+} from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+import { Requester } from 'sonarwhal/dist/src/lib/connectors/utils/requester';
+import { URL } from 'url';
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class NoBrokenLinksRule implements IRule {
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.accessibility,
+            description: `Rule to flag broken links in the page`
+        },
+        id: 'no-broken-links',
+        schema: [],
+        scope: RuleScope.site
+    };
+
+    public constructor(context: RuleContext) {
+        const requester = new Requester();
+        const brokenStatusCodes = [404, 410, 500, 503];
+
+
+        // This method checks whether a url is broken and report if it is
+        const reportBrokenLink = async (hrefUrl: string, resource: any, linkElement: IAsyncHTMLElement) => {
+            const absoluteUrl = new URL(hrefUrl, resource).href;
+
+            await requester.get(absoluteUrl).then(
+                async (value) => {
+                    const statusIndex = brokenStatusCodes.indexOf(value.response.statusCode);
+
+                    if (statusIndex>-1) {
+                        await context.report(
+                            absoluteUrl,
+                            linkElement,
+                            `Broken link found (${brokenStatusCodes[statusIndex]} response)`
+                        );
+                    }
+                },
+                async (reason) => {
+                    debug(`Error accessing {$absoluteUrl}. ${JSON.stringify(reason)}`);
+                    await context.report(
+                        absoluteUrl,
+                        linkElement,
+                        'Broken link found (404 response)'
+                    );
+                }
+            );
+        };
+
+        const checkImageElement = async(linkElement: IAsyncHTMLElement, resource: any) => {
+
+            const imageSrc = linkElement.getAttribute('src');
+
+            await reportBrokenLink(imageSrc, resource, linkElement);
+
+            // Check for srcset attribute value if that exist
+            const srcSet = linkElement.getAttribute('srcset');
+
+            if (srcSet!==null) {
+                const sources = srcSet.split(',').filter((x) => {
+                    return x!=='';
+                });
+
+                for (const source of sources) {
+                    const url = source.split(' ').filter((x) => {
+                        return x!=='';
```

Comment:
`x !== ''`

Reply:
Fixed now. Interestingly, the linter did not catch this!

Reply:
> Fixed now. Interestingly, the linter did not catch this!

I don't think we set ESList to check for it.

@kshyju Wanna look into and update [`.eslintrc.json`](https://github.com/sonarwhal/sonarwhal/blob/e9623e3c2a09f259d6fc6fb5bbd2fcb029a51151/.eslintrc.json#L17) to make ESLint check for that? Thanks!